submissions/balancer-design/universal_8to8.md

```diff
@@ -0,0 +1,68 @@
+
+![Broken auto-looping 4-4](https://cdn.discordapp.com/attachments/699618487097884723/813450086234587156/unknown.png)
+
+To remedy this, we need to make sure every item on the loop can get to every single one of the loop's exits. The way to do this with the least splitters is a secondary balancer:
+
+![Fixed 4-4](https://cdn.discordapp.com/attachments/699618487097884723/813450357639872553/unknown.png)
+
+And we're pretty much done with the 4-4. It can be improved, but more on that later. Now, let's try our hand at something bigger: a universal 8-8. We do the same steps as before, and get this:
+
+![Basic universal 8-8](https://cdn.discordapp.com/attachments/699618487097884723/813450635667832842/unknown.png)
+
+However, there is an arrangement in which this happens:
+
+![Basic 8-8 bottlenecking](https://cdn.discordapp.com/attachments/699618487097884723/813450849481130012/unknown.png)
+
+6 belts are available, but less than 6 are coming out. This is a problem. The standard 8-8 balancer is throughput-limited, meaning the entire thing is as well. Luckily, this is easily fixed by simply replacing the core balancer with a throughput-unlimited one. Notice how the loopback can stay throughput-limited:
+
+![Proper universal 8-8](https://cdn.discordapp.com/attachments/699618487097884723/813451151625289778/unknown.png)
+
+Now comes the time for the usual procedure of moving stuff around until we achieve black hole-level component densities. Because we used blue belts, we can employ belt weaving to get the 8 belts of loopback to be 4 tiles wide. Here is the finished balancer:
+
+![Compacted 8-8 universal](https://cdn.discordapp.com/attachments/699618487097884723/813451640144658432/unknown.png)
+
+It is unrecognizable, but that's because the secondary balancer is now inline with the main one, meaning we have to deal with up to 12 belts shoved into a 8 tile wide space, which forces the secondary balancer to be extremely spread out.
+
+### Final note
+
+We don't actually need as many belts of loopback as we used. In general, for an N-N universal balancer (where N is a power of 2 like 4, 8, 16 etc) we need N-3 belts looping back. While I could have done that, the extra mass of splitters to merge 8 to 5 and then back to 8 would have made it take even more space. However, this fact is really helpful for the 4-4 universal, because it allows us to have only 2 belts of loopback:
+
+![Simpler universal 4-4](https://cdn.discordapp.com/attachments/699618487097884723/813451928238948372/unknown.png)
+
+Then after some compacting, we get a design comically small in comparison to the 8-8:
+
+![Compact universal 4-4 next to universal 8-8](https://cdn.discordapp.com/attachments/699618487097884723/813452464120528896/unknown.png)
+
+It's important to note that this universal 4-4 is not my own design, although it's based on a series of balancers I made long ago. Unfortunately, I don't remember who created it.
```

Comment:
You should end off just reiterating what you created. And then explain what the reader should do with this information. For example: Should he use it in his base? Should he try to design his own? Is it more a thought experiment? If you want to use it in your base, in what situation is it useful? Has this design been done before for an 8-8?